cmd/dep/init.go

```diff
@@ -239,6 +242,75 @@ func hasImportPathPrefix(s, prefix string) bool {
 	return strings.HasPrefix(s, prefix+"/")
 }
 
+// itemizedFeedback itemizes resolved dependencies and logs the solution as
+// feedback.
+func itemizedFeedback(m *dep.Manifest, l *dep.Lock, pkgT pkgtree.PackageTree) {
+	// Algo:
+	// 1. collect direct external deps from PackageTree
+	// 2. loop through lock projects
+	//   2.a. check if the projects exists in direct deps list
+	//   2.b. if it's a direct dep:
+	//     2.b.i. if it has a version attached, constraint dep
+	//     2.b.ii. if it has revision only, hint dep
+	//   2.c. else it's a transitive dep
+	//   2.d. log feedback
+
+	// Extract direct external deps
+	rm, errmap := pkgT.ToReachMap(true, true, false, nil)
+	for fail := range errmap {
+		internal.Logf("Warning: %s", fail)
+	}
+
+	directDeps := rm.Flatten(false)
+
+	for _, lock := range l.Projects() {
+		// Get locked revision, version and branch
+		r, v, b := gps.VersionComponentStrings(lock.Version())
+
+		// Check if it's a valid SHA1 digest and trim to 7 characters.
+		if len(r) == 40 {
+			if _, err := hex.DecodeString(r); err == nil {
+				// Valid SHA1 digest
+				r = r[0:7]
+			}
+		}
+
+		// Get LockedVersion
+		var ver string
+		if v != "" {
+			ver = v
+		} else if b != "" {
+			ver = b
+		}
+
+		cf := &internal.ConstraintFeedback{
+			LockedVersion: ver,
+			Revision:      r,
+			ProjectPath:   string(lock.Ident().ProjectRoot),
+		}
+
+		// Get manifest version if available
+		if pp, ok := m.Dependencies[lock.Ident().ProjectRoot]; ok {
+			cf.Version = pp.Constraint.String()
```

Comment:
I'm not sure. And unfortunately, I don't _think_ there's a way to see those errors. Would be a great thing to open an issue for 😄